packages/web-components/src/components/masthead/megamenu-category-link.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import DDSLink from '../../globals/internal/link';
+import styles from './masthead.scss';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * MegaMenu category sublink
+ *
+ * @element dds-megamenu-category-link
+ */
+@customElement(`${ddsPrefix}-megamenu-category-link`)
+class DDSMegaMenuCateoryLink extends DDSLink {
+  /**
+   * link title.
+   */
+  @property({ reflect: true })
+  title = '';
+
+  /**
+   * autoid.
+   */
+  @property({ reflect: true })
+  autoid = '';
+
+  /**
+   * index for data-autoid.
+   */
+  @property({ reflect: true })
+  index = 0;
+
+  connectedCallback() {
+    if (!this.hasAttribute('tabindex')) {
+      this.setAttribute('tabindex', '0');
+    }
+    if (!this.hasAttribute('data-autoid')) {
+      const { autoid, index } = this;
+      this.setAttribute('data-autoid', `${autoid}-item${index}`);
+    }
```

Comment:
Given `data-autoid` is a transient notion that helps migration from our React library, etc., we don't set it automatically in component code. That said, we can simply let our adaptors use `data-autoid` instead of using `autoid` and `index`.